### PR TITLE
feat: add a component to support rate limt control

### LINKED ISSUE #20: Feature: Add a component  to support RateTime Control
### Checklist

- [X] I've searched for similar issues and couldn't find anything matching
- [X] I've discussed this feature request in the [OpenIMSDK Slack](https://join.slack.com/t/openimsdk/shared_invite/zt-1tmoj26uf-_FDy3dowVHBiGvLk9e5Xkg) and got positive feedback

### Is this feature request related to a problem?

❎ No

### Problem Description

none

### Solution Description

use redis + lua 

### Benefits

Add a component to support RateTime Control in order to make API more safe and maintainbale

### Potential Drawbacks

none

### Additional Information

none

### PR DESCRIPTION
**Pull request by AllianceTing**
</br>

#### 👀 What this PR does / why we need it:

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed

#### 🅰 Which issue(s) this PR fixes:

Fixes #20

### CODE REVIEW

**Comment by cubxxw** [comment]
not:
```
Fixes issue20
```
but:
```
Fixes #20
```
example: Fixes #20

**Comment by cubxxw** [comment]
not:
```
feat: issue20 add a component to support rate limt control
```
but:
```
feat: add a component to support rate limt control
```

File: server/internal/middleware/frequencyControl.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+func LimitHandler(maxConn int, refreshRate float64) gin.HandlerFunc {
+	tb := &frequencyControlByTokenBucket{
+		capacity:    int64(maxConn),
+		refreshRate: refreshRate,
+		tokens:      0,
+		lastToken:   time.Now(),
+	}
+	return func(c *gin.Context) {
+		if !tb.Allow() {
+			c.String(503, "Too many request")
```
Comment (IRONICBo) [inline server/internal/middleware/frequencyControl.go:49]: use constant in `http` or define a constant in `OpenKF/server/internal/common/code.go`.

Comment (AllianceTing) [reply server/internal/middleware/frequencyControl.go]: ok i'll solve it

**Comment by IRONICBo** [comment]
@AllianceTing  Please check CI report https://github.com/OpenIMSDK/OpenKF/actions/runs/5418577637/jobs/9850806240?pr=29

**Review by AllianceTing — COMMENTED** [review]
format coding style

**Review by AllianceTing — COMMENTED** [review]
fix bug

**Review by cubxxw — CHANGES_REQUESTED** [review]

**Pull request merged**